forgerock-openbanking-model/src/main/java/com/forgerock/openbanking/model/SoftwareStatement.java

```diff
@@ -50,7 +51,21 @@
     @LastModifiedDate
     public Date updated;
     private String applicationId;
-    private String organisationId;
+
+
+    /**
+     * The software ID is the same as the ForgeRock Application Id. It is included here to make logic simpler for
+     * TPPS wishing to support message signing with both ForgeRock certificates and OB issued certificates. OB issued
+     * Software Statements contain software_id and org_id fields and specify that these values should be used in the
+     * iss field of a jws header (message signing header). Having both the software_id and org_id fields in the
+     * ForgeRock software statement will make TPP and our own test logic easier when performing message signing.
+     */
+    private String software_id;
+    /**
+     * The organisation ID - this is the organisation in the ForgeRock directory under which this
+     * application/software id has been created.
+     */
+    private String org_id;
```

Comment:
Really good javadoc 👍 

The only shame is the inconsistent format of the variable names compared to others in this class. I don't understand why you need the underscores, or why organisation is abbreviated?

Reply:
Because I'm trying to match the fields an Open Banking issued Software Statement;

Payload of one of those has software_id and org_id. This makes it easier for TPPs to do message signing. We can have the same code to pull these fields from the Software Statement when createding a JWS;
```
{
  "org_jwks_endpoint": "TODO",
  "software_mode": "TEST",
  "software_redirect_uris": [],
  "org_status": "Active",
  "software_client_id": "5f97e32724e4990010afec13",
  "iss": "ForgeRock",
  "software_jwks_endpoint": "https://service.directory.dev-ob.forgerock.financial:8074/api/software-statement/5f97e32724e4990010afec13/application/jwk_uri",
  "software_id": "5f97e32724e4990010afec13",
  "org_contacts": [],
  "ob_registry_tos": "https://directory.dev-ob.forgerock.financial:8074/tos/",
  "org_id": "5f905b2e9638120010f83168",
  "software_jwks_revoked_endpoint": "TODO",
  "software_roles": [
    "AISP",
    "CBPII",
    "PISP",
    "DATA"
  ],
  "exp": 1604394409,
  "org_name": "Anonymous",
  "org_jwks_revoked_endpoint": "TODO",
  "iat": 1603789609,
  "jti": "ba921964-27cb-43b9-8b60-b615fc3c0e97"
}
```

Reply:
Then we end up with this in our api-tests;

```
    fun getDetachedJws(body: Any, tpp: Tpp, version: OBVersion): String {
        val softwareStatement = JWSObject.parse(tpp.registrationResponse.software_statement);
        val ssPayload = softwareStatement.payload.toJSONObject();
        val orgId: String = ssPayload.get("org_id") as String;
```

Reply:
And we don't have to care if the Software Statement was issued by FR or OB directory

Reply:
If I've understood that correctly, it's because you want to serialise/deserialise the class as JSON and have it match the OB spec?

In that case, you could use the `@JsonProperty` annotation:

```
@JsonProperty("org_id")
private String organisationId;
```

Reply:
Ah I see you've merged. Oh well, never mind.